test/jasmine/tests/bar_test.js

```diff
@@ -1801,14 +1801,6 @@ function mockBarPlot(dataWithoutTraceType, layout) {
     supplyAllDefaults(gd);
     Plots.doCalcdata(gd);
 
-    var plotinfo = {
-        xaxis: gd._fullLayout.xaxis,
-        yaxis: gd._fullLayout.yaxis
-    };
-
-    // call Bar.crossTraceCalc
-    Bar.crossTraceCalc(gd, plotinfo);
```

Comment:
⚠️ ⚠️ ⚠️  Removed because this is included in `doCalcdata` now, but in fact the test FAILS if we leave this in. That means `Bar.crossTraceCalc` is not idempotent, though I didn't really look into what it's doing. I don't think this is particularly a problem, but it might indicate it's more fragile than it should be.

Reply:
> That means Bar.crossTraceCalc is not idempotent, though I didn't really look into what it's doing.

Yep, I noticed that last week. This commit https://github.com/plotly/plotly.js/pull/2860/commits/dfada6ac7e144696d4edc42c33f1754d331cc93d helps, but there are more cases where `Bar.setPositions` oops I mean `Bar.crossTraceCalc` mutates `gd.calcdata`.